src/backend/distributed/utils/citus_clauses.c

```diff
@@ -29,6 +29,32 @@ static Expr * citus_evaluate_expr(Expr *expr, Oid result_type, int32 result_typm
 
 
 /*
+ * Whether the executor needs to reparse and try to execute this query.
+ */
+bool
+RequiresMasterEvaluation(Query *query)
+{
+	ListCell *targetEntryCell = NULL;
+
+	foreach(targetEntryCell, query->targetList)
```

Comment:
This seemed like the cleanest code, the naive alternative is to have ErrorIfModifyQueryNotSupported somehow signal that this query will need evaluation but that's too ugly. I admittedly don't have performance numbers but think this will only add real planning time to more complicated queries, and those will take a while to run anyway?